app/main.py

```diff
@@ -2,0 +1,19 @@
+from app.errors import (
+    VaccineError,
+    NotWearingMaskError
+)
+from app.cafe import Cafe
+
+
+def go_to_cafe(friends: list[dict], cafe: Cafe) -> str:
+    count_of_mask = 0
+    for friend in friends:
+        try:
+            cafe.visit_cafe(friend)
+        except VaccineError:
+            return "All friends should be vaccinated"
+        except NotWearingMaskError:
+            count_of_mask += 1
```

Comment:
Code Efficiency: You don't need to use a counter for friends without masks. You can return the message right away when you catch a 'NotWearingMaskError'. This way, you won't have to iterate through all friends when one of them is not wearing a mask.